recipe/thirdparty/providers/custom.go

```diff
@@ -61,20 +61,20 @@ func NewProvider(input tpmodels.ProviderInput) tpmodels.TypeProvider {
 
 		if tenantId == tpmodels.DefaultTenantId {
 			// if tenantId is default, merge the client configs
-			staticClientConfigs := []tpmodels.ProviderClientConfig{}
-			copy(staticClientConfigs, input.Config.Clients)
+			configToReturn.Clients = []tpmodels.ProviderClientConfig{}
+			copy(configToReturn.Clients, input.Config.Clients) // making a copy so that I don't change the original input.Config.Clients array
```

Comment:
this doesn't make sense.. cause configToReturn points to input.Config anyway. So you are making a change to input.Config as well here..